eip712/messages.py

```diff
@@ -13,14 +13,17 @@
 from eip712.hashing import hash_domain
 from eip712.hashing import hash_message as hash_eip712_message
 
+# ! Do not change the order of the fields in this list !
+# To correctly encode and hash the domain fields, they
+# must be in this precise order.
 EIP712_DOMAIN_FIELDS = [
     "name",
     "version",
     "chainId",
     "verifyingContract",
     "salt",
 ]
-HEADER_FIELDS = set(f"_{field}_" for field in EIP712_DOMAIN_FIELDS)
+HEADER_FIELDS = [f"_{field}_" for field in EIP712_DOMAIN_FIELDS]
```

Comment:
set is unordered, which doesn't work for us, since we want our HEADER_FIELDS to follow precisely the EIP712_DOMAIN_FIELDS ordering